### PR TITLE
☁️ Add Cloudflare Logs Integration

### PR DESCRIPTION
**Pull request by shaeqahmed**
\
**Issue**: #89 

Implemented a Cloudfare managed log source with support for the following tables:

- audit
- dns
- firewall_event
- http_request
- nel_report
- network_analytics
- spectrum_event

#### Demo

Screenshot 2023-02-25 at 10 39 27 PM

### CODE REVIEW

File: data/managed/log_sources/cloudflare/tables/audit.yml [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  - cloud.instance.id
+  - cloud.instance.name
+  - cloud.machine.type
+  - cloud.project.id
+  - cloud.provider
+  - cloud.region
+  - container.id
+  - container.image.name
+  - container.labels
+  - container.name
```
Comment (Samrose-Ahmed) [inline data/managed/log_sources/cloudflare/tables/audit.yml:16]: `cloud`, `container`, `host` are all unused, remove ?

File: data/managed/log_sources/cloudflare/tables/audit.yml [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+  if .user.id != null { 
+    .related.user = push(.related.user, .user.id) 
+  }
+
+  if .source.ip != null { 
+    .related.ip = push(.related.ip, .source.ip) 
+  }
+
+  # if .tags == null || !includes(array!(.tags), "preserve_duplicate_custom_fields") {
```
Comment (Samrose-Ahmed) [inline data/managed/log_sources/cloudflare/tables/audit.yml:182]: kept for a reason or?

File: data/managed/log_sources/cloudflare/log_source.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    } else if match(.__metadata.s3.key, r'http_request') {
+      "http_request"
+    } else if match(.__metadata.s3.key, r'nel_report') {
+      "nel_report"
+    } else if match(.__metadata.s3.key, r'network_analytics') {
+      "network_analytics"
+    } else if match(.__metadata.s3.key, r'spectrum_event') {
+      "spectrum_event"
+    } else {
+      "default" # will error table not found
```
Comment (Samrose-Ahmed) [inline data/managed/log_sources/cloudflare/log_source.yml:20]: why err and not abort, e.g. will it cause error when Cloudflare sends verification file ? : https://developers.cloudflare.com/logs/get-started/enable-destinations/aws-s3/#manage-via-the-cloudflare-dashboard

**Review by Samrose-Ahmed — COMMENTED** [review]
gg, minor comments

File: data/managed/log_sources/cloudflare/tables/audit.yml (hunk 1 above)
Comment (shaeqahmed) [reply data/managed/log_sources/cloudflare/tables/audit.yml:16]: Sure lets remove

File: data/managed/log_sources/cloudflare/log_source.yml (hunk 3 above)
Comment (shaeqahmed) [reply data/managed/log_sources/cloudflare/log_source.yml:20]: lets abort

File: data/managed/log_sources/cloudflare/tables/audit.yml (hunk 2 above)
Comment (shaeqahmed) [reply data/managed/log_sources/cloudflare/tables/audit.yml:182]: idk, we can think abt if an option like this is necessary. should we always just delete those duplicate fields if not, or always keep?

**Pull request merged**